README.md

```diff
@@ -56,18 +56,36 @@ $ python melted_UKBB_extract.py --config-file myconfig.yaml --data-file current.
 
 ### Use inside python
 
-The function `extract_UKBB_tabular_data` takes in a `dict` of key-value pairs containing the same entries as the `YAML`
-config file, as well as the paths to the input files:
+The function `extract_UKBB_tabular_data` has the following signature:
 
 ```python
-def extract_UKBB_tabular_data(config=None, data_file=None, dictionary_file=None, coding_file=None, verbose=False)
+def extract_UKBB_tabular_data(
+    config: Config,
+    data_file: str | None = None,
+    dictionary_file: str | None = None,
+    coding_file: str | None = None,
+    verbose: str | None = False,
+) -> tuple[pl.DataFrame, pl.DataFrame | None, pl.DataFrame, pl.DataFrame]:
 ```
 
-The function returns 3 or 4 polars DataFrames depending on the `config['wide']` setting:
+The return signature depends on the `config['wide']` setting. Unfortunately, as Python < 3.11 does not support `TypedDict` with generic types, this cannot be (easily) expressed with Python hints. Once Python 3.10 is no longer supported, this could be updated with a generic type.
```

Comment:
Just to be clear here, TypedDict is supported, but not the full set of features we would like to use <3.11?